deployments/intersmash-deployments-provider/src/main/java/org/jboss/intersmash/deployments/WildflyDeploymentApplicationConfiguration.java

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.intersmash.deployments;
+
+import com.google.common.base.Strings;
+
+import cz.xtf.core.config.XTFConfig;
+
+/**
+ * Defines the contract for WILDFLY application services that expose configuration which should be taken into account
+ * by the WILDFLY s2i v2 process, i.e. the WildFly Maven plugin, feature packs and BOMs settings
+ */
+public interface WildflyDeploymentApplicationConfiguration {
+
+	String WILDFLY_EE_FEATURE_PACK_LOCATION = "wildfly.ee-feature-pack.location";
+	String WILDFLY_FEATURE_PACK_LOCATION = "wildfly.feature-pack.location";
+	String WILDFLY_CLOUD_FEATURE_PACK_LOCATION = "wildfly.cloud-feature-pack.location";
+	String WILDFLY_EE_CHANNEL_GROUPID = "wildfly.ee-channel.groupId";
+	String WILDFLY_EE_CHANNEL_ARTIFACTID = "wildfly.ee-channel.artifactId";
+	String WILDFLY_EE_CHANNEL_VERSION = "wildfly.ee-channel.version";
+	String WILDFLY_BOMS_EE_SERVER_VERSION = "bom.wildfly-ee.version";
+	String WILDFLY_MAVEN_PLUGIN_GROUPID = "wildfly-maven-plugin.groupId";
+	String WILDFLY_MAVEN_PLUGIN_ARTIFACTID = "wildfly-maven-plugin.artifactId";
+	String WILDFLY_MAVEN_PLUGIN_VERSION = "wildfly-maven-plugin.version";
+	String MAVEN_MIRROR_URL = "maven-mirror.url";
+
+	default String wildflyMavenPluginGroupId() {
+		return XTFConfig.get(WILDFLY_MAVEN_PLUGIN_GROUPID);
+	}
+
+	default String wildflyMavenPluginArtifactId() {
+		return XTFConfig.get(WILDFLY_MAVEN_PLUGIN_ARTIFACTID);
+	}
+
+	default String wildflyMavenPluginVersion() {
+		return XTFConfig.get(WILDFLY_MAVEN_PLUGIN_VERSION);
+	}
+
+	default String eeFeaturePackLocation() {
+		return XTFConfig.get(WILDFLY_EE_FEATURE_PACK_LOCATION);
+	}
```

Comment:
nice idea: IIUC, the former `String eeFeaturePackLocation();` now becomes a default method...

Reply:
yup, it is just a configuration interface that behaves in the Intersmash documented way, and we use it easily this way, without the need for overriding that stuff each time.